### ISSUE TITLE
Large batch_size leads to inaccurate validation loss calculation

### ISSUE THREAD

**Issue by felixlaumon** [dnouri/nolearn#247]
Related code: https://github.com/dnouri/nolearn/blob/master/nolearn/lasagne/base.py#L593-L608

The training and validation losses are aggregated across batches. However under the following circumstances:

1. `batch_size` is very large or the number of batches is small
3. losses fluctuates hugely across the dataset
4. the last batch has very few samples

Aggregating across batches will be quite different as across samples (because the last batch with few samples will be weighted higher).

Obviously this is an edge case and usually happens only in the first few epoch, but I think it's quite important to have the losses calculated correctly.

**Comment by cancan101** [comment]
[`numpy.average` has an option to take weights. ](http://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.average.html#numpy.average). Perhaps keep track of batch sizes and use that for average.

**Comment by felixlaumon** [comment]
Didn't know `numpy.average` exists and I was about implemented something similar myself. I will send out a PR later today.
